### DISCUSSION TITLE
Ignoring blanks in csv files

### DISCUSSION THREAD

**Discussion by ThinkTransit** [django-import-export/django-import-export#1737] [Q&A] [answered]
Hi,

Any thoughts on the best way to handle empty lines in csv files when importing? Sometimes when converting excel to csv there are empty lines at the bottom of the file that need to deleted manually.

For example 
```
,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
```

I think a simple check in before_import_row could maybe work, if all fields are empty ignore row.

Any better ideas? Would a PR to include this functionality + config option be considered?

Thanks

**Comment by matthewhegarty** [comment, marked as answer]
There is a similar question [here](https://github.com/django-import-export/django-import-export/issues/1192#issue-718848641).  Like you say, I would recommend override `before_import` or `before_import_row` to sanitize the data.  I would consider a PR to enable this with a setting on the resource for the v4 release (see release-4 branch).  Alternatively documenting best practice would be ok.

**Reply by ThinkTransit** [reply]
Thanks @matthewhegarty sounds good. I will attempt to put a PR in soon, an override would work fine however it would be nice to enable globally.

**Reply by matthewhegarty** [reply]
Thanks - contributing guidelines are [here](https://django-import-export.readthedocs.io/en/latest/contributing.html).  This feature should be disabled by default, and documentation provided on how to enable.
